src/firecracker/src/main.rs

```diff
@@ -96,6 +108,19 @@ fn main_exec() -> Result<(), MainError> {
     #[cfg(target_arch = "aarch64")]
     enable_ssbd_mitigation();
 
+    if let Err(err) = resize_fdtable() {
+        match err {
+            // These errors are non-critical: In the worst case we have worse snapshot restore
+            // performance.
+            ResizeFdTableError::GetRlimit | ResizeFdTableError::Dup2(_) => {
+                logger::warn!("Failed to resize fdtable: {err}")
```

Comment:
IMO, printing a warning for an optimization is a bit too much. I'd keep this a debug message

Reply:
Fair, will change